src/kubernetes.cc

```diff
@@ -737,8 +754,8 @@ void KubernetesReader::WatchMaster(
     // A notification for watch completion.
     std::mutex completion_mutex;
     std::unique_lock<std::mutex> watch_completion(completion_mutex);
-    Watcher watcher(callback, std::move(watch_completion),
-                    config_.VerboseLogging());
+    Watcher watcher(std::bind(&EventCallback, callback, std::placeholders::_1),
```

Comment:
`std::placeholders::_N` leaves me a bit puzzled. Is there a way of doing these methods without using unbound arguments?

Reply:
They confuse me too. Unfortunately, there is no better way.
If it helps, think about `std::bind` as: you have to list all of the arguments to the bound function, but some of those can represent the arguments to the function being called -- that's what placeholders are (`std::placeholders::_1` is the first argument to the resulting function).